tools/kubectl-ownerref/main.go

```diff
@@ -5,60 +5,175 @@ import (
+		fallback: rest.NewWarningWriter(os.Stderr, rest.WarningWriterOptions{Deduplicate: true}),
+	}
+	kubeConfigFlags.WrapConfigFn = func(c *rest.Config) *rest.Config {
+		c.WarningHandler = warningHandler
+		c.QPS = 100
+		return c
 	}
 
-	flag.Parse()
+	matchVersionKubeConfigFlags := cmdutil.NewMatchVersionFlags(kubeConfigFlags)
+
+	f := cmdutil.NewFactory(matchVersionKubeConfigFlags)
+
+	ioStreams := genericclioptions.IOStreams{In: os.Stdin, Out: os.Stdout, ErrOut: os.Stderr}
 
-	if err := run(kubeconfig); err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
+	rootCommand := NewCmdGetByOwner("kubectl", f, ioStreams)
+	rootCommand.PersistentFlags().AddGoFlagSet(flag.CommandLine)
+	matchVersionKubeConfigFlags.AddFlags(rootCommand.PersistentFlags())
+	kubeConfigFlags.AddFlags(rootCommand.PersistentFlags())
+
+	if err := rootCommand.Execute(); err != nil {
 		os.Exit(1)
 	}
+}
+
+type GetByOwnerOptions struct {
+	PrintFlags *get.PrintFlags
+	ToPrinter  func(*meta.RESTMapping, *bool, bool, bool) (printers.ResourcePrinterFunc, error)
+
+	Namespace         string
+	ExplicitNamespace bool
+
+	AllNamespaces bool
+
+	OwnerKindArg string
+	OwnerKind    schema.GroupKind
+	OwnerName    string
 
+	genericclioptions.IOStreams
 }
 
-func run(kubeconfig *string) error {
-	if len(os.Args) < 3 {
-		fmt.Println("Please complete command: kubectl ownerref [KIND] [NAME]")
-		os.Exit(1)
+type warningHandler struct {
+	fallback rest.WarningHandler
+}
+
+func (h *warningHandler) HandleWarningHeader(code int, agent string, text string) {
+	if code == 299 {
+		// Ignore deprecated kind messages, the user isn't normally specifying the kinds
+		return
+	}
+	h.fallback.HandleWarningHeader(code, agent, text)
+}
+
+// NewGetByOwnerOptions returns a GetByOwnerOptions with default options.
+func NewGetByOwnerOptions(parent string, streams genericclioptions.IOStreams) *GetByOwnerOptions {
+	return &GetByOwnerOptions{
+		PrintFlags: get.NewGetPrintFlags(),
+
+		IOStreams: streams,
+	}
+}
+
+func NewCmdGetByOwner(parent string, f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewGetByOwnerOptions(parent, streams)
+
+	cmd := &cobra.Command{
+		Use: "kubectl ownerref",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, cmd, args))
+			cmdutil.CheckErr(o.Validate(cmd))
+			cmdutil.CheckErr(o.Run(f, cmd))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	cmd.Flags().BoolVarP(&o.AllNamespaces, "all-namespaces", "A", o.AllNamespaces, "If present, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+	return cmd
+}
+
+// Complete takes the command arguments and factory and infers any remaining options.
+func (o *GetByOwnerOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+
+	if len(args) == 2 {
+		o.OwnerKindArg = args[0]
+		o.OwnerName = args[1]
+	} else {
+		return fmt.Errorf("syntax: kubectl ownerref [KIND] [NAME]")
```

Comment:
In command line format, I think `[ ]` means optional and `< >` is better? :thinking: 

```suggestion
		return fmt.Errorf("syntax: kubectl ownerref <KIND> <NAME>")
```